src/lib/utils/SynapseContext.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useContext } from 'react'
+import { QueryClient, QueryClientProvider } from 'react-query'
+
+const defaultQueryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      staleTime: 30 * 1000, // 30s
+      retry: false, // SynapseClient knows which queries to retry
+    },
+  },
+})
+
+export type SynapseContextType = {
+  accessToken?: string
+  isInExperimentalMode: boolean
+  utcTime: boolean
+}
+
+export const SynapseContext = React.createContext<
+  SynapseContextType | undefined
+>(undefined)
```

Comment:
This needs to be exported to use it in class components.

TODO: Add comment stating the same